### ISSUE TITLE
[bug] Replace failed node replaces incorrect failed node

### ISSUE THREAD

**Issue by br0nd** [zwave-js/zwavejs2mqtt#834]
### Version

**Checklist:**

- [X] I am **not** using HomeAssistant. **Or:** a developer has told me to come here.
- [X] I have checked the troubleshooting section and my problem is **not** described there.
- [X] I have read the changelog problem was **not** mentioned there.

Build/Run method

- [X] Docker
- [ ] PKG
- [ ] Manually built (git clone - npm install - npm run build )

zwavejs2mqtt version: 2.1.1
zwave-js version: 6.6.0

### Describe the bug

Replacing failed node replaces incorrect failed node

### To Reproduce

1. Go to "Failed Node"  (Node 51 in my case)
2. Click on "Replace Failed Node"
3. Put replacement device in Include Mode (Zooz Zen 26 in my case)
4. Look at logs (or Control Panel Table)
5. Notice that replacement switch was included but replaced the wrong failed node (in my case it replaced node 54 instead of node 51)

### Expected behavior

New Zwave device would be included to replace the original node id.. not a different failed node id

### Additional context
I did not capture detailed logs (I can try to recreate the issue and capture debug logs)

16:15:02.838 CNTRLR   starting replace failed node process...
16:15:02.838 CNTRLR » [Node 051] pinging the node...
16:15:03.610 CNTRLR   [Node 051] Node 51 did not respond after 1 attempts, it is presumed dead
16:15:03.613 CNTRLR   [Node 051] ping failed: Failed to send the command after 1 attempts (Status No Ack)
16:15:03.770 CNTRLR   handling replace node request (status = FailedNodeReplace)
16:15:03.771 CNTRLR   The failed node is ready to be replaced, inclusion started...
16:15:32.838 CNTRLR   stopping inclusion process...
16:15:32.897 CNTRLR   handling add node request (status = Done)
16:15:32.897 CNTRLR   done called for 54
16:15:32.898 CNTRLR   the inclusion process was stopped
16:16:06.393 CNTRLR « [Node 054] Received updated node info

**Comment by robertsLando** [comment]
@br0nd Do you have full zwavejs logs? @AlCalzone That's so strange...

**Comment by AlCalzone** [comment]
@br0nd Was the node you were trying to include reset to factory defaults?
Otherwise it could be that it still had the ID 54 stored and did not accept the ID 51 which the controller tried to give it.

**Comment by br0nd** [comment]
> @br0nd Was the node you were trying to include reset to factory defaults?
> Otherwise it could be that it still had the ID 54 stored and did not accept the ID 51 which the controller tried to give it.

Unlikely, but possible, here is some more detail that could help
TLDR - Node 51 was a working switched, I firmware upgraded it and it became dead, I tried to replace failed node with the same switch (after firmware upgrade) and first try it made the container crash, 2nd time it replaced the wrong node.

I do have another Zooz Zen21 I need to firmware upgrade, I can put both log levels to silly.. and see if I can recreate the same behavior again.. or close this issue. Let me know if there is anything specific you want me to do before I retry with a different switch.

Additional details
1. Node 51 was a working functional Zooz Zen21 On/Off switch
2.  I used zwavejs2mqtt to upgrade Zen21 firmware
3. After upgrade, Node 51 reported as dead (but the switch was still functioning). I researched and determined that this is a known limitation with Zooz switches after firmware upgrade. You have to re-include them
4.  I did not factory reset this switch, but I assume firmware upgrade does this..? I guess its possible some identifier in this firmware update made it look like Node 54 (Node 54 is another failed node, but it was originally a Zooz Zen27)?
5. I did the replace failed node for node 51 in zwavejs2mqtt and put the switch in inclusion mode and the container crashed 
6. I restarted the docker container, and did replace node 51 again and put the switch in inclusion mode again.. this time the replace node was successful but for Node 54

**Comment by br0nd** [comment]
I tried to firmware update Node 50 (another Zooz Zen21 switch), and while the firmware update was successful, when I went to try replace failed node, the container crashed.

I have included two sets of log files (container log, general log and zwave log) for both the firmware update (which was successful) and the replace failed node (which crashed the container). 
[Logs.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6103831/Logs.zip)


I'm also including the snippet showing the crash of the container in the container log..

{"log":"\u001b[90m2021-03-08 10:35:06.124\u001b[39m \u001b[32mINFO\u001b[39m \u001b[1mZWAVE\u001b[22m: Success zwave api call replaceFailedNode { success: true, message: 'Success zwave api call', result: true }\n","stream":"stdout","time":"2021-03-08T17:35:06.125056447Z"}
{"log":"\u001b[90m2021-03-08 10:35:13.129\u001b[39m \u001b[31mERROR\u001b[39m \u001b[1mZWAVE-SERVER\u001b[22m: Unexpected error The message cannot be sent because node 50 is dead\n","stream":"stdout","time":"2021-03-08T17:35:13.129951405Z"}
{"log":"ZWaveError: The message cannot be sent because node 50 is dead\n","stream":"stdout","time":"2021-03-08T17:35:13.130002911Z"}
{"log":"    at Driver.sendMessage (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:2035:12)\n","stream":"stdout","time":"2021-03-08T17:35:13.130011125Z"}
{"log":"    at Driver.sendCommand (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:2195:17)\n","stream":"stdout","time":"2021-03-08T17:35:13.130015732Z"}
{"log":"    at BinarySwitchCCAPI.set (/usr/src/app/node_modules/zwave-js/src/lib/commandclass/BinarySwitchCC.ts:111:3)\n","stream":"stdout","time":"2021-03-08T17:35:13.130019785Z"}
{"log":"    at Proxy.BinarySwitchCCAPI.\u003ccomputed\u003e (/usr/src/app/node_modules/zwave-js/src/lib/commandclass/BinarySwitchCC.ts:124:3)\n","stream":"stdout","time":"2021-03-08T17:35:13.130024435Z"}
{"log":"    at ZWaveNode.setValue (/usr/src/app/node_modules/zwave-js/src/lib/node/Node.ts:693:4)\n","stream":"stdout","time":"2021-03-08T17:35:13.130028962Z"}
{"log":"    at Function.handle (/usr/src/app/node_modules/@zwave-js/server/dist/lib/node/message_handler.js:15:33)\n","stream":"stdout","time":"2021-03-08T17:35:13.130032902Z"}
{"log":"    at Client.receiveMessage (/usr/src/app/node_modules/@zwave-js/server/dist/lib/server.js:85:62)\n","stream":"stdout","time":"2021-03-08T17:35:13.130036723Z"}
{"log":"\u001b[90m2021-03-08 10:35:19.203\u001b[39m \u001b[32mINFO\u001b[39m \u001b[1mZWAVE\u001b[22m: Controller status: Inclusion stopped\n","stream":"stdout","time":"2021-03-08T17:35:19.204599163Z"}
{"log":"\u001b[90m2021-03-08 10:35:19.229\u001b[39m \u001b[32mINFO\u001b[39m \u001b[1mZWAVE\u001b[22m: Node 50: removed\n","stream":"stdout","time":"2021-03-08T17:35:19.229987573Z"}
{"log":"\u001b[90m2021-03-08 10:35:19.230\u001b[39m \u001b[32mINFO\u001b[39m \u001b[1mZWAVE\u001b[22m: Node removed 50\n","stream":"stdout","time":"2021-03-08T17:35:19.231177203Z"}
{"log":"\u001b[90m2021-03-08 10:35:22.515\u001b[39m \u001b[32mINFO\u001b[39m \u001b[1mZWAVE\u001b[22m: Node 50: added\n","stream":"stdout","time":"2021-03-08T17:35:22.515995599Z"}
{"log":"\u001b[90m2021-03-08 10:35:22.516\u001b[39m \u001b[34mDEBUG\u001b[39m \u001b[1mZWAVE\u001b[22m: Node 50 has been added to nodes array\n","stream":"stdout","time":"2021-03-08T17:35:22.517222055Z"}
{"log":"\u001b[90m2021-03-08 10:35:22.517\u001b[39m \u001b[34mDEBUG\u001b[39m \u001b[1mZWAVE\u001b[22m: Binding to node 50 events\n","stream":"stdout","time":"2021-03-08T17:35:22.518027293Z"}
**[{"log":"This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:\n","stream":"stdout","time":"2021-03-08T17:35:22.523341136Z"}
{"log":"TypeError: Cannot read property 'basic' of undefined\n","stream":"stdout","time":"2021-03-08T17:35:22.523483034Z"}
{"log":"    at Object.dumpDeviceClass (/usr/src/app/node_modules/@zwave-js/server/dist/lib/state.js:85:26)\n","stream":"stdout","time":"2021-03-08T17:35:22.523499456Z"}
{"log":"    at Object.dumpNode (/usr/src/app/node_modules/@zwave-js/server/dist/lib/state.js:42:67)\n","stream":"stdout","time":"2021-03-08T17:35:22.523506549Z"}
{"log":"    at /usr/src/app/node_modules/@zwave-js/server/dist/lib/forward.js:23:31\n","stream":"stdout","time":"2021-03-08T17:35:22.523512219Z"}
{"log":"    at Array.forEach (\u003canonymous\u003e)\n","stream":"stdout","time":"2021-03-08T17:35:22.523517125Z"}
{"log":"    at ZWaveController.\u003canonymous\u003e (/usr/src/app/node_modules/@zwave-js/server/dist/lib/forward.js:20:34)\n","stream":"stdout","time":"2021-03-08T17:35:22.523522246Z"}
{"log":"    at ZWaveController.emit (node:events:390:22)\n","stream":"stdout","time":"2021-03-08T17:35:22.523527285Z"}
{"log":"    at ZWaveController.EventEmitter.emit (node:domain:470:12)\n","stream":"stdout","time":"2021-03-08T17:35:22.523532252Z"}
{"log":"    at ZWaveController.handleReplaceNodeRequest (/usr/src/app/node_modules/zwave-js/src/lib/controller/Controller.ts:1203:11)\n","stream":"stdout","time":"2021-03-08T17:35:22.523536973Z"}
{"log":"    at Driver.handleRequest (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1936:22)\n","stream":"stdout","time":"2021-03-08T17:35:22.523541952Z"}
{"log":"    at Driver.handleUnsolicitedMessage (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1670:5)\n","stream":"stdout","time":"2021-03-08T17:35:22.523555581Z"}])**

**Comment by AlCalzone** [comment]
The crash is this: https://github.com/zwave-js/zwave-js-server/issues/149 which has been fixed.

Also in your latest log, Node 50 gets replaced with 50 - so that seemed to work. I can only assume that your stick was in a weird state where both Node 51 and 54 were the same node - although I don't know how that could happen.

Generally during inclusion, the controller tells the node "you're Node XYZ now", which the node accepts unless it still has a node ID.
A firmware update should not change that ID - maybe the node just needed a little time. Hard to tell without a log of the problem.

**Comment by br0nd** [comment]
Thank you, you are right that Node 50 was successfully replaced, no idea what happened last time. And I have updated docker to get the new version of zwave-js-server. I will log a new issue if I see the container crashing or replacing the wrong node id.

Thanks for your help!
